packages/ra-ui-materialui/src/layout/Theme/ThemeProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { ReactNode, useCallback, useEffect, useRef, useState } from 'react';
+
+import {
+    createTheme,
+    ThemeProvider as MuiThemeProvider,
+    Theme,
+} from '@mui/material/styles';
+import { DeprecatedThemeOptions } from '@mui/material';
+import { ThemeContext, ThemeSetter } from './ThemeContext';
+
+export const ThemeProvider = ({
```

Comment:
Please add JSDoc for usage

Reply:
This is almost an internal component. Do you want an example that would be a copy of `react-admin` `Admin` component?

Reply:
No, that's fine.